src/esperanza/walletextension.h

```diff
@@ -202,6 +202,45 @@ class WalletExtension : public staking::StakingWallet {
       const finalization::VoteRecord &vote2);
 
   void PostInitProcess(CScheduler &scheduler);
+
+  template <typename Callable>
+  void ForEachMatureOutput(
+      const CTransaction &tx,
+      const CBlockIndex *block,
+      blockchain::Depth depth,
+      Callable f) const {
+    std::size_t reward_offset = 0;
+    if (tx.IsCoinBase() && !block) {
+      // Coinbase transaction is not in the main chain
+      return;
+    }
+    if (tx.IsCoinBase() && depth <= COINBASE_MATURITY) {
+      reward_offset = m_dependencies.GetFinalizationRewardLogic().GetNumberOfRewardOutputs(block->nHeight) + 1;
+    }
+    for (std::size_t i = reward_offset; i < tx.vout.size(); ++i) {
+      f(tx.vout[i], i);
+    }
+  }
+
+  template <typename Callable>
+  void ForEachOutputWithMaturity(
```

Comment:
I think I would appreciate a comment on both `ForEachMatureOutput` and `ForEachOutputWithMaturity` which especially explains what is the distinction between these two (they sound suspiciously similar)

Reply:
Added comments.